src/python/LASSO.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn import linear_model
+from scipy.stats import pearsonr
+from sklearn.metrics import mean_squared_error
+from sklearn.metrics import mean_absolute_error
+from sklearn.model_selection import KFold
+from sklearn.linear_model import LassoCV
+from sklearn.linear_model import MultiTaskLassoCV
+from yellowbrick.datasets import load_concrete
+from yellowbrick.regressor import AlphaSelection
+from yellowbrick.regressor.alphas import alphas
+
+
+
+# ### Merge datasets based on cancer cell line name;
+# ### concatenate the two dataframes and match based on cancer cell lines; 
+# ### save them into two new dataframes- MET and GCP; 
+# ### split GCP and MET into train and test set respectively.
+# 
+
+# In[15]:
+
+
+md = pd.read_excel(io='CCLE metabolomics dataset.xlsx',sheet_name="All")
+mt = md.drop(['Tissue', 'Medium','Culture'], axis=1)
+hm = pd.read_csv('GCP_proteomics_remapped.csv')
+merge_tb = mt.merge(hm,how='inner',left_on='CCL', right_on='Cell Line')
+MET = merge_tb.iloc[:,1:226]
+GCP = merge_tb.iloc[:,227:269]
+
+GCP = np.nan_to_num(GCP, nan=0)
+MET = np.nan_to_num(MET, nan=0)
+
+Xtrain, Xtest, Ytrain, Ytest =train_test_split(GCP, MET, test_size=0.3, random_state=0)
+print(Xtest.shape)
+print(Ytest.shape)
+print(Xtrain.shape)
+print(Ytrain.shape[1])
+
+
+
+# ### Tuning the alpha
+
+# In[4]:
+
+
+
+alphas = [10**-6, 10**-5,10**-4,10**-3,10**-2,10**-1,1,2,3,4,5,6,7,8,9,10]
+
+mdl = list()
+error = list()
+for i,a in enumerate(alphas):
+    for j in range(Ytrain.shape[1]):
+        x=linear_model.Lasso(alpha=a).fit(Xtrain, Ytrain[:, j])
+        mdl.append(x)
+    ypred = mdl[i].predict(Xtest)
+    error.append(mean_squared_error(ypred,Ytest[:, j]))
```

Comment:
Same thing here for the same exact reason.

Reply:
Then to save on memory, write some code to update your model list. The logic is the following:

1. Evaluate models with first alpha
2. On the next iteration, save your models and errors in another object
3. Write code that will see if the MSE in the current alpha level is less than the MSE in the previous alpha level
4. If it is, replace the model object in your original list, and update the alpha associated with this model
5. Otherwise, don't update and continue